### ISSUE TITLE
Blur is not triggered on Firefox

### ISSUE THREAD

**Issue by vinyll** [yabwe/medium-editor#1161]
### Description

The `blur` event callback is called on Chromium ()

### Steps to reproduce

See https://gist.github.com/vinyll/16fa34cdfa146fb035245da596e0a04c

**Expected behavior:** [What you expected to happen]

When editing text and then clicking outside the console should log _'blurred!'_

**Actual behavior:** [What actually happened]

It behaves as expected on Chromium (51) but not on Firefox (47 nor 49)

### Versions

- medium-editor: latest
- browser: ff 47 ff 49
- OS: MacOSX

**Comment by FiodorGherasimenco** [comment]
Instead 

```
  editor.subscribe('blur', function (event, editable) {
    console.debug('blurred!', event, editable)
  })
```

Try to use 

```
  editor.subscribe('editableBlur', function (event, editable) {
    console.debug('blurred!', event, editable)
  })
```

**Comment by vinyll** [comment]
I had tried that but it triggers the 'editableBlur' also when clicking on the editor buttons.
The 'blur' event is exactly what I expect though it does not work with Firefox

**Comment by FiodorGherasimenco** [comment]
Hm.. what i saw in your html you don have the body but the handle some events on then body. 

```
this.attachDOMEvent(this.options.ownerDocument.body, 'mousedown', this.handleBodyMousedown.bind(this), true);
this.attachDOMEvent(this.options.ownerDocument.body, 'click', this.handleBodyClick.bind(this), true);
this.attachDOMEvent(this.options.ownerDocument.body, 'focus', this.handleBodyFocus.bind(this), true);
```
 try to remove body  in medium js so the event will be on this.options.ownerDocument.

i got same problem with placeholder when you focus in and then click on body placeholder stays hidden

**Comment by nmielnik** [comment]
@vinyll it appears the root of this issue is that there isn't anything else in your document to click on.  I was actually able to get this to reproduce in chrome as well if you ensure there's nothing else within the page and click outside of the 'body' of the document.

You can see `blur` fire properly if you just simply add another element into your page:

```html
<meta charset=utf-8>
<script src="http://cdn.jsdelivr.net/medium-editor/latest/js/medium-editor.min.js"></script>
<link rel="stylesheet" href="http://cdn.jsdelivr.net/medium-editor/latest/css/medium-editor.min.css" type="text/css">

<section class=editable>
  Test me
</section>
<div>something else</div> <!-- Added another element -->
<script>
  var editor = new MediumEditor('.editable')
  editor.subscribe('blur', function (event, editable) {
    console.debug('blurred!', event, editable)
  })
</script>
```

If there isn't another element on the page to click on or for focus to move to, it appears the combination of `mousedown`, `click`, and `focus` on the body aren't sufficient.

However, it seems pretty unlikely that someone would only have a single editable div on the page and absolutely nothing else, so I'm not super concerned about this bug, but I'll leave the issue open in case someone wants to investigate, or if someone finds another case where this is a problem.

**Comment by vinyll** [comment]
I confirm it works when clicking on another element of the page.
Not the highest priority bug indeed. It's mostly confusing when trying the editor or edge cases like sizing the _body_ of the page and focusing out of it.

**Comment by nmielnik** [comment]
Yeah, agreed, it can be a bit confusing for some of the more common testing scenarios.  Thanks for opening the issue, perhaps someone will try to figure out a simple fix for this...

**Comment by ulion** [comment]
I encounter similar case, when call blur() on the editor element, the subscribed blur event is not triggered. The reason I think is that medium editor does not handle raw blur event of the element, instead, it "guesses" blur event by some other "focus" events, which caused it missed many cases when the blur does happen but no "focus" event captured.

**Comment by ulion** [comment]
Browser tab switch should also trigger blur, but here it does not.
